### DISCUSSION TITLE
Não consigo escrever arquivo com encoding correto

### DISCUSSION THREAD

**Discussion by angellicadearaujo** [frontendbr/forum#180] [Discussão]
Olá, pessoal
Estou com problemas para escrever o conteúdo de uma variável em um arquivo texto. Eu acredito que seja problema de encoding/charset. 

Alguém já trabalhou com essa API? Alguém pode me dar uma luz? Segue Gist:

https://gist.github.com/angellicacardozo/79fd5fbd1774f0762a43967ccaf35413

**Comment by felipefialho** [comment]
Olá @angellicacardozo, tomei a liberdade de arrumar seu titulo para facilitar a leitura.

@fdaciuk e @diogomoretti, tem alguma ideia?

**Comment by fdaciuk** [comment]
@angellicacardozo tá tentando fazer isso no browser mesmo? Não cheguei a mexer muito com isso, mas a File System API ainda está bem instável ainda, e tem uma grande probabilidade de não ser aprovada.

Não sei também se você está testando só para estudar, ou se vai usar em algum projeto real, mas é importante ter isso em mente =)

Onde você vai precisar usar isso? Pergunto porque pode ter soluções melhores para fazer o que você precisa xD

Só para ter como referência:
[Especificação pela W3C](https://www.w3.org/TR/file-system-api/)
[Documentação na MDN](https://developer.mozilla.org/en-US/docs/Web/API/File_System_API)

**Comment by angellicadearaujo** [comment]
@fdaciuk teoricamente eu não tenho escolha. Eu reconheço que a API é instável e pode não ser aprovada mas é o que eu tive que fazer no curto prazo. Você conhece alguma alternativa?

**Comment by danguilherme** [comment]
A API não somente é instável como não é um padrão ([e nem vai ser](http://www.html5rocks.com/en/tutorials/file/filesystem/)): http://lists.w3.org/Archives/Public/public-webapps/2014AprJun/0010.html

**Comment by fdaciuk** [comment]
@angellicacardozo para esse caso, não vai ter como fugir do backend. Tem várias formas de fazer. Vou colocar algumas alternativas:
1. Você pode fazer seu próprio backend, como uma API, e, via frontend, você envia os dados para o backend e salva o arquivo. Se precisar que o usuário tenha esse arquivo localmente, dê uma opção de ele fazer download;
   - O envio pode ser: ou via request http (ajax), ou via socket.
2. Se não quiser fazer o backend, você pode utilizar um Database as a Service por exemplo, para salvar essa informação. Mas para gerar um arquivo formatado, e dar opção do usuário baixar, não vai ter como fugir de fazer seu próprio backend para tratar isso.

Deve ter milhões de formas de se fazer isso, mas essas duas alternativas podem ajudar =)
